### PR TITLE
Add FixityCheck, create it after characterization, delete it with object

### PR DESCRIPTION
**Pull request by hackmastera**
refs #142
for the rest of the plan, see https://github.com/pulibrary/figgy/issues/142#issuecomment-361742266

### CODE REVIEW

File: app/change_set_persisters/plum_change_set_persister/delete_fixity_check.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    def initialize(change_set_persister:, change_set:)
+      @change_set = change_set
+      @change_set_persister = change_set_persister
+    end
+
+    def run
+      return unless resource.respond_to? :original_file
+      resource.original_file.file_identifiers.each do |file_id| 
+        query_service.custom_queries.find_by_string_property(property: :file_id, value: file_id).
+          select{ |obj| obj.class == FixityCheck }.each do |fc|
```
Comment (tpendragon) [inline app/change_set_persisters/plum_change_set_persister/delete_fixity_check.rb]: Should we add an optional `model:` parameter to this query?

File: app/change_set_persisters/plum_change_set_persister/delete_fixity_check.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+class PlumChangeSetPersister
+  class DeleteFixityCheck
```
Comment (tpendragon) [inline app/change_set_persisters/plum_change_set_persister/delete_fixity_check.rb]: Maybe we should just store the fixity properties on the FileSet - maybe in a nested object to keep them nicely contained, if that matters? That way we don't have to worry about this - the fixity check will just be attached to the life of the fileset.

Comment (hackmastera) [reply app/change_set_persisters/plum_change_set_persister/delete_fixity_check.rb]: if it's nested doesn't that means it doesn't get its own created / updated dates? we are going to need those.

Comment (hackmastera) [reply app/change_set_persisters/plum_change_set_persister/delete_fixity_check.rb]: well updated, at least. I guess that could be managed manually.

**Comment by hackmastera** [comment]
superceded by #799

**Pull request closed without merging**